qcodes/utils/dataset/doNd.py

```diff
@@ -286,8 +285,7 @@ def do1d(
             datasaver.add_result(
                 (param_set, set_point), *call_param_meas(), *additional_setpoints_data
             )
-
-    param_set.post_delay = original_delay
+            sleep(delay)
```

Comment:
One potential optimization that we could do is to get the time before and after calling add_result and only sleep the delay - the time needed to call add_result that way we would save a bit of time if add_result is slow

Reply:
That's a good point. We can substract add_result time from the delay and sleep with the remaining time.